python/hail/keytable.py

```diff
@@ -332,6 +332,44 @@ def join(self, right, how='inner'):
         return KeyTable(self.hc, self._jkt.join(right._jkt, how))
 
     @handle_py4j
+    def distinct_by_key(self):
+        """Return a new key table whose keys are the distinct keys of this key
+        table by keeping one row among all rows with a duplicate key.
+        The row that is kept is chosen arbitrarily.  Requires a
+        shuffle.
+
+        :return: Key table with keys the distinct keys of this key table
+        :rtype: :class:`.KeyTable`
+
+        """
+        
+        return KeyTable(self.hc, self._jkt.distinctByKey())
+
+    @handle_py4j
+    @typecheck_method(right=kt_type)
+    def broadcast_left_join_distinct(self, right):
```

Comment:
I think it's not a good idea to add methods with the same functionality but different implementation. I would prefer to add a flag onto join - left and inner are both easy (with and without distinct), and we can error out on unsupported modes.

Reply:
This is a good idea.

I can't say I'm super jazzed by `...join(..., broadcast_right=True, right_make_distinct=True)` when we have a `distinct` operation on `KeyTable`s.  Maybe I won't handle distinct and we'll leave it for the query optimizer.

Reply:
Yeah, I think the join shouldn't have a distinct option. It should do a cartesian product like the others.